### PR TITLE
Bound the argument to Number.toFixed

### PR DESCRIPTION
**Pull request by embooglement**
The argument to `Number.prototype.toFixed` is required to be between 0 and 20, otherwise it will throw a `RangeError`. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed

This PR explicitly clamps the argument to that range.

### CODE REVIEW

**Comment by embooglement** [comment]
@Addepar/pants

File: numeral.js [hunk 1]
```diff
@@ -54,10 +54,15 @@
           boundedPrecision = minDecimals;
         }
 
+        // Number.toFixed requires it's argument be between 0 and 20, inclusive.
+        // See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed
+        boundedPrecision = Math.max(Math.min(boundedPrecision, 20), 0);
```
Comment (sicasli) [inline numeral.js]: can `boundedPrecision` be negative initially? if so, we need test to cover that.

Comment (embooglement) [inline numeral.js]: I don't think so. `toFixed` is called in 3 places:

When there are optional decimal parts (`0,0.0000[000]`):

```
d = toFixed(value, (precision[0].length + precision[1].length), roundingFunction, precision[1].length);
```

Here precision is an array of strings, so the lengths are always non-negative, and `precision[1].length <= precision[0].length + precision[1].length`, so `minDecimals <= maxDecimals`

When there are no optional parts, but there is a decimal part (`0,0.0000`):

```
d = toFixed(value, precision.length, roundingFunction);
```

Here `precision` is a string. This sets `minDecimals` to `0`, and it must be that `maxDecimals >= 0`

And when there is no decimal part (`0,0`):

```
w = toFixed(value, null, roundingFunction);
```

Here `boundedPrecision` is set to `minDecimals` which will be `0`.

Comment (embooglement) [inline numeral.js]: I do need to add a test for the optional case though, didn't see that case before.

Comment (sicasli) [inline numeral.js]: so you don't have to do `Math.max( --- , 0)` then

Comment (embooglement) [inline numeral.js]: True. Will remove.

**Comment by embooglement** [comment]
@Addepar/pants @sicasli This is good to go.

File: tests/numeral/format.js [hunk 2]
```diff
@@ -87,7 +87,10 @@ exports.format = {
                 [-5444333222111, '0,0 aT~', '-5'],
                 [123456, '0.0[0] aK~', '123.46'],
 
-            ],
+                // toFixed precision is never out of bounds
+                [1, '0,0.000000000000000000000', '1.00000000000000000000'],
+                [1.111111111111111111111, '0,0.000000000000000000[000]', '1.111111111111111160']
```
Comment (sicasli) [inline tests/numeral/format.js]: Worth adding one more test with format `0,0.00[0000000000000000000]`

**Comment by sicasli** [comment]
lgtm

File: tests/numeral/format.js (hunk 2 above)
Comment (embooglement) [inline tests/numeral/format.js]: Done

**Pull request merged**